### ISSUE TITLE
Feature suggestion: include Lombok in SpringSTS

### ISSUE THREAD

**Issue by danbeck** [spring-projects/sts4#543]
Lombok is used by many projects; it makes coding with java much easier.

I would like to make the suggestion to include Lombok in the eclipse STS distribution.

**Comment by kdvolder** [comment]
We could do this, but I am reluctant. Yes, I agree, Lombok is useful and a nice thing to have. And yes, many people use it.

However, even if many people use it, not everyone does. For people who do not use Lombok there is no benefit to having it installed but there are a few drawbacks. 

To me the biggest potential drawback is on occasion Lombok breaks existing functionality. I have personally experienced this, installing lombok and basically my STS/Eclipse stopped working completely. Sometimes the breakage may be more subtle. 

While this may not be a common occurrence, (in fact I installed and used Lombok yesterday into STS and it worked just fine). It does mean that by including Lombok we are essentially causing potential issues for people who do not even want to use it Lombok.

For that reason I'm inclined to stick with our decision not to include lombok by default in our distribution, but ask that people who want to use it install it themselves.

That being said, maybe its worth also to discuss this with the rest of the team. Having used lombok recently without any problems my position on this has somewhat mellowed and if others on the team feel the same... perhaps...

**Comment by martinlippert** [comment]
@danbeck First of all, thanks a lot for the suggestion and the feature request. They are very welcome.

In this concrete case, I don't think we will include Lombok into he Eclipse distribution of the Spring Tools by default. We experimented with this quite a bit in the past and even had an internal build that included Lombok by default for some time, but we came to the conclusion that we will move ahead with that.

Kris mentioned a few reasons for this above. Lombok integrates very deeply with Eclipse and modifies core pieces of the Eclipse Java support. This makes a lot of sense from the Lombok support point of view, but it also requires very careful updates and double checking for each and every update to the Eclipse JDT implementation. I am sure the Lombok project does its best to support each and every Eclipse JDT update and version, but it is a tough job to keep up with the changes and to guarantee at the same time that the original Eclipse JDT implementation continues to work as without the modifications from Lombok.

At the same time, we ship a new version of the Spring Tools for Eclipse on top of the latest release, in most cases on the same day or week of the Eclipse release itself - to always provide our users with the latest Eclipse releases, including latest versions of the Java support. Including the Lombok support would mean that we would require a released version of Lombok that guarantees to work with the latest Eclipse release basically on the same day than the Eclipse release itself. Realistically, I don't expect that to happen anytime soon. Therefore we decided to not include Lombok by default.

It is always a hard and very careful decision on our side what to include in the default distribution and what to leave out of it and in the hand of our users to install manually. Our goal is to provide a ready-to-use distribution for as many of our users as possible, but that sadly also means to say no to specific extensions from time to time.

This is not a decision made forever and might change in the future. We are happy to re-evaluate the integration into the default distribution sometime in the future, but given the nature of the Lombok integration into Eclipse, something fundamentally would need to change for that.

Sorry to reject your suggestion. Hope you understand the reasons for it.

Beyond that, if there is anything we could do to make the manual installation process for users who would like to use Lombok in their IDE installs easier, we would be more than happy to take that into account.
